app/client/components/billing/billing.tsx

```diff
@@ -31,13 +37,27 @@ import { PaymentDetailsTable } from "../payment/paymentDetailsTable";
 import { PaymentFailureAlertIfApplicable } from "../payment/paymentFailureAlertIfApplicable";
 import { ErrorIcon } from "../svgs/errorIcon";
 import { GiftIcon } from "../svgs/giftIcon";
+import { InvoicesTable } from "./invoicesTable";
 
 type MMACategoryToProductDetails = {
   [mmaCategory in GroupedProductTypeKeys]: ProductDetail[];
 };
 
-const BillingRenderer = (apiResponse: MembersDataApiItem[]) => {
-  const allProductDetails = apiResponse.filter(isProduct).sort(sortByJoinDate);
+class BillingDataAsyncLoader extends AsyncLoader<
+  [MembersDataApiItem[], { invoices: InvoiceDataApiItem[] }]
+> {}
+
+const BillingRenderer = (
+  apiResponse: [MembersDataApiItem[], { invoices: InvoiceDataApiItem[] }]
```

Comment:
how about de-structuring (by name) to avoid the `apiResponse[0]` and `apiResponse[1]` further down...
```suggestion
  [mdaResponse, invoiceResponse]: [MembersDataApiItem[], { invoices: InvoiceDataApiItem[] }]
```